extensions/amp-form/0.1/form-proxy.js

```diff
@@ -0,0 +1,281 @@
+/**
+ * Copyright 2016 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+/**
+ * @param {!HTMLFormElement} form
+ * @return {!Object}
+ */
+export function installFormProxy(form) {
+  const proxy = {};
```

Comment:
Can we have a proxy prototype (that we initialize on the first form?

The proxy instance would then just have a private field like `instance_` and the getters would access the `form` through `this.instance_`